integration/management-api-tests/src/test/java/org/wso2/micro/integrator/api/TemplatesResourceTestCase.java

```diff
@@ -1,91 +1,70 @@
- *"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
- *KIND, either express or implied.  See the License for the
- *specific language governing permissions and limitations
- *under the License.
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
  */
 
 package org.wso2.micro.integrator.api;
 
-import org.apache.http.HttpResponse;
-import org.awaitility.Awaitility;
 import org.json.JSONObject;
 import org.testng.Assert;
 import org.testng.annotations.AfterClass;
 import org.testng.annotations.BeforeClass;
 import org.testng.annotations.Test;
 import org.wso2.esb.integration.common.utils.ESBIntegrationTest;
-import org.wso2.esb.integration.common.utils.clients.SimpleHttpClient;
 
 import java.io.IOException;
-import java.util.HashMap;
-import java.util.Map;
-import java.util.concurrent.TimeUnit;
+
+import static org.wso2.micro.integrator.api.Constants.COUNT;
+import static org.wso2.micro.integrator.api.Constants.LIST;
 
 public class TemplatesResourceTestCase extends ESBIntegrationTest {
 
+    private String accessToken;
+    private String endpoint;
+
     @BeforeClass(alwaysRun = true)
     public void setEnvironment() throws Exception {
         super.init();
+        accessToken = TokenUtil.getAccessToken(hostName, portOffset);
+        endpoint = "https://" + hostName + ":" + (DEFAULT_INTERNAL_API_HTTPS_PORT + portOffset) + "/management/templates";
     }
 
-    @Test(groups = { "wso2.esb" }, description = "Test get Templates resource")
+    @Test(groups = {"wso2.esb"}, description = "Test get Templates resource")
     public void retrieveTemplates() throws IOException {
-
-        String responsePayload = sendHttpRequestAndGetPayload(null);
+        String responsePayload = sendHttpRequestAndGetPayload(endpoint, accessToken);
         JSONObject jsonResponse = new JSONObject(responsePayload);
-        Assert.assertEquals(jsonResponse.get("count"), 2, "Assert Failed due to the mismatch of " +
+        Assert.assertEquals(jsonResponse.get(COUNT), 2, "Assert Failed due to the mismatch of " +
                 "actual vs expected resource count");
-        Assert.assertTrue(jsonResponse.get("list").toString().contains("AbcTemplate"), "Assert failed " +
+        Assert.assertTrue(jsonResponse.get(LIST).toString().contains("AbcTemplate"), "Assert failed " +
                 "since expected resource name not found in the list");
-        Assert.assertTrue(jsonResponse.get("list").toString().contains("testSequenceTemplate"), "Assert failed " +
+        Assert.assertTrue(jsonResponse.get(LIST).toString().contains("testSequenceTemplate"), "Assert failed " +
                 "since expected resource name not found in the list");
```

Comment:
Since this is repetitive in other test classes as well, Shall we try to refactor it further?

Reply:
yeah seems it's better. done